src/leveled_sst.erl

```diff
@@ -2950,37 +2960,44 @@ key_dominates_expanded([H1|T1], [H2|T2]) ->
 %%% Timing Functions
 %%%============================================================================
 
--spec update_buildtimings(
-    erlang:timestamp(), build_timings(), atom()) -> build_timings().
+-spec update_buildtimings(build_timings(), atom()) -> build_timings().
 %% @doc
 %%
 %% Timings taken from the build of a SST file.
 %%
 %% There is no sample window, but the no_timing status is still used for
 %% level zero files where we're not breaking down the build time in this way.
-update_buildtimings(_SW, no_timing, _Stage) ->
+update_buildtimings(no_timing, _Stage) ->
     no_timing;
-update_buildtimings(SW, Timings, Stage) ->
-    Timer = timer:now_diff(os:timestamp(), SW),
+update_buildtimings(Timings, Stage) ->
+    LastTS = Timings#build_timings.last_timestamp,
+    ThisTS = os:timestamp(),
```

Comment:
up to taste, but would it not be nicer to say here: `NewTimings` = Timings#build_timings{last_timestamp = os:timestamp()}` and then in all cases below use `NewTimings#build_timings{...}` where you only update the thing that matters, not the timestamp.

Reply:
Fixed